pkg/cvo/cvo_scenarios_test.go

```diff
@@ -2856,6 +2862,111 @@ func TestCVO_UpgradePreconditionFailingAcceptedRisks(t *testing.T) {
+				{Type: configv1.RetrievedUpdates, Status: configv1.ConditionFalse},
+			},
+		},
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	defer shutdownFn()
+
+	// make the image report unverified
+	payloadErr := &payload.UpdateError{
+		Reason:  "ImageVerificationFailed",
+		Message: "The update cannot be verified: some random error",
+		Nested:  fmt.Errorf("some random error"),
+	}
+	if !isImageVerificationError(payloadErr) {
+		t.Fatal("not the correct error type")
+	}
+	worker := o.configSync.(*SyncWorker)
+	retriever := worker.retriever.(*fakeDirectoryRetriever)
+	retriever.Set(PayloadInfo{}, payloadErr)
+	retriever.Set(PayloadInfo{Directory: "testdata/payloadtest-2", Verified: true}, nil)
+
+	go worker.Start(ctx, 1)
+
+	// Step 1: Simulate a verified payload being retrieved and ensure the operator sets verified
+	//
+	client.ClearActions()
+	err := o.sync(ctx, o.queueKey())
+	if err != nil {
+		t.Fatal(err)
+	}
+	actions := client.Actions()
+	if len(actions) != 2 {
+		t.Fatalf("%s", spew.Sdump(actions))
+	}
+	expectGet(t, actions[0], "clusterversions", "", "version")
+	expectUpdateStatus(t, actions[1], "clusterversions", "", &configv1.ClusterVersion{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "version",
+			ResourceVersion: "1",
+			Generation:      1,
+		},
+		Spec: configv1.ClusterVersionSpec{
+			ClusterID:     clusterUID,
+			Channel:       "fast",
+			DesiredUpdate: &configv1.Update{Version: desired.Version, Image: desired.Image},
+		},
+		Status: configv1.ClusterVersionStatus{
+			ObservedGeneration: 1,
+			// Prefers the operator's version
+			Desired:     configv1.Release{Version: o.release.Version, Image: o.release.Image},
+			VersionHash: "DL-FFQ2Uem8=",
+			History: []configv1.UpdateHistory{
+				{State: configv1.CompletedUpdate, Image: "image/image:0", Version: "1.0.0-abc", Verified: true, StartedTime: defaultStartedTime, CompletionTime: &defaultCompletionTime},
+			},
+			//Capabilities is zero as upgrade is not accepted
+			Conditions: []configv1.ClusterOperatorStatusCondition{
+				{Type: ImplicitlyEnabledCapabilities, Status: "False", Reason: "AsExpected", Message: "Capabilities match configured spec"},
+				{Type: configv1.OperatorAvailable, Status: configv1.ConditionTrue, Message: "Done applying 1.0.0-abc"},
+				// cleared failing status and set progressing
+				{Type: ClusterStatusFailing, Status: configv1.ConditionFalse},
+				{Type: configv1.OperatorProgressing, Status: configv1.ConditionFalse, Message: "Cluster version is 1.0.0-abc"},
+				{Type: configv1.RetrievedUpdates, Status: configv1.ConditionFalse},
+				{Type: DesiredReleaseAccepted, Status: configv1.ConditionFalse, Reason: "VerifyPayloadVersion",
+					Message: `Verifying payload failed version="1.0.0-abc" image="image/image:0" failure=release image version 1.0.1-abc does not match the expected upstream version 1.0.0-abc`},
```

Comment:
I'm not clear on where this `1.0.1-abc` vs. `1.0.0-abc` divergence comes from.  From what I understand, the test case is setting `image/image:0` up as `1.0.0-abc`, but this line is claiming it's `1.0.1-abc`?

Reply:
Did a little digging:
The line for the msg is:
https://github.com/openshift/cluster-version-operator/blob/f0139d19c34dbddd4a59148d472aa44d468660b3/pkg/cvo/sync_worker.go#L352

It can be traced back from the input of the case:
https://github.com/openshift/cluster-version-operator/pull/1081/files#diff-6cca1696ca6d83b25f5baf586b99602d32827c05c4276dd03ebfcb6825ff2558R2866
and eventually from
https://github.com/openshift/cluster-version-operator/blob/58356deee675c900b7cd4742ceefc0e6a84d7efb/pkg/cvo/testdata/payloadtest-2/release-manifests/release-metadata

The setup of this new case was duplicated from https://github.com/openshift/cluster-version-operator/blob/f0139d19c34dbddd4a59148d472aa44d468660b3/pkg/cvo/cvo_scenarios_test.go#L2859 with `worker.stillInitializingFunc = func() bool { return false }` removed which is for covering the introduced branch.

Reply:
Maybe https://github.com/openshift/cluster-version-operator/blob/58356deee675c900b7cd4742ceefc0e6a84d7efb/pkg/cvo/testdata/payloadtest/release-manifests/release-metadata makes more sense now?

Let me give it a try.